### PR TITLE
#10793 junit5

### LINKED ISSUE #10793: Update to JUnit5
### Problem Description
JUnit5 is out since 2017 and brings the following benefits:

* we get more granularity, and can import only what's necessary
* JUnit 5 allows multiple runners to work simultaneously
* JUnit5 allows Java8+ language features like lambda functions, making tests more powerful and easier to maintain.
* new features for describing, organizing, and executing tests. For instance, tests get better display names and can be organized hierarchically (e.g. https://medium.com/@boonkeat/android-unit-testing-with-junit5-d1b8f9c620b6)
* Allows combination of multiple extensions

Furthermore JUnit5 gives us more freedom of choice and comfort related to replacing our CDI unit testing framework.
See https://github.com/hzi-braunschweig/SORMAS-Project/issues/10074#issuecomment-1294691128

### Proposed Change
Update to JUnit5. This means replacing imports and few clean ups.
The whole process should be done in one step.
Check whether Android app should be updated as-well or separately. -> Separately

### Implementation Details

- [x] org.unit.Test -> org.junit.jupiter.api.Test
- [x] org.junit.Assert.* -> org.junit.jupiter.api.Assertions.*
- [x] org.junit.Assert.assertThat -> org.hamcrest.MatcherAssert.assertThat
- [x] Before -> BeforeEach
- [x] BeforeClass -> BeforeAll
- [x] After -> AfterEach
- [x] ArchUnitRunner no longer needed
- [x] Test(expected=Exception) -> assertThrows
- [x] Some assertions currently have string message first -> is last param instead now
- [x] Use latest WireMock version for Junit5
- [x] CaseFacadeEjbTest: Extract wire mock related tests
- [x] TestContainers Junit5 extension + workaround: https://github.com/testcontainers/testcontainers-java/issues/970#issuecomment-625044008
- [x] Move tests for external surveillance functionality from CaseServiceTest and EventServiceTest into ExternalSurveillanceToolGatewayFacadeEjbTest
- [x] Remove sharing logic from testArchive... test in CaseFacadeEjbTest and EventFacadeEjbTest

### PR DESCRIPTION
**Pull request by MartinWahnschaffe**
Fixes #10793

- org.unit.Test -> org.junit.jupiter.api.Test
- org.junit.Assert.* -> org.junit.jupiter.api.Assertions.* org.junit.Assert.assertThat -> org.hamcrest.MatcherAssert.assertThat Before -> BeforeEach
- BeforeClass -> BeforeAll
- After -> AfterEach
- ArchUnitRunner no longer needed
- Test(expected=Exception) -> assertThrows
- Some assertions currently have string message first -> is last param instead now Use latest WireMock version for Junit5
- CaseFacadeEjbTest: Extract wire mock related tests TestContainers Junit5 extension + workaround: https://github.com/testcontainers/testcontainers-java/issues/970#issuecomment-625044008 Move tests for external surveillance functionality from CaseServiceTest and EventServiceTest into ExternalSurveillanceToolGatewayFacadeEjbTest Remove sharing logic from testArchive... test in CaseFacadeEjbTest and EventFacadeEjbTest

It's a lot of classes, but 80-90% of the changes follow the same pattern. When reviewing the most important part is to make sure that I didn't accidentally remove relevant parts from the tests.

### CODE REVIEW

**Comment by StefanKock** [comment]
First finding: What about replacing JUnit in `sormas-app`?

**Comment by MartinWahnschaffe** [comment]
See issues description -> out of scope for this issue. The dependencies are separate anyway, so there is no need to include it here.

File: sormas-ui/src/test/java/de/symeda/sormas/ui/importer/CountryImporterTest.java [hunk 1]
```diff
@@ -29,7 +27,7 @@
 
 // Using Silent Runner to ignore unnecessary stubbing exception
 // which is a side effect of extending AbstractBeanTest
-@RunWith(MockitoJUnitRunner.Silent.class)
+// @RunWith(MockitoJUnitRunner.Silent.class)
```
Comment (StefanKock) [inline sormas-ui/src/test/java/de/symeda/sormas/ui/importer/CountryImporterTest.java:30]: Why is this commented and not removed?

**Review by StefanKock — APPROVED** [review]
I reviewed all 209! changed files, and only found it curious that two annotations where commented out. If this is clarified; i would say let's quickly merge this and inform the team (probable compile errors imminent).

**Pull request merged**